nixos/modules/services/web-servers/nginx/default.nix

```
@@ -254,20 +254,20 @@ let
             else defaultListen;
 
         listenString = { addr, port, ssl, extraParameters ? [], ... }:
-          "listen ${addr}:${toString port} "
-          + optionalString ssl "ssl "
+          (if ssl && vhost.http3 then "
+          # UDP listener for **QUIC+HTTP/3
+          listen ${addr}:${toString port} http3 "
+          + optionalString vhost.default "default_server "
+          + optionalString vhost.http3Reuseport "reuseport"
+          + ";" else "")
```

```suggestion
          (lib.optional (ssl && vhost.http3) "
          # UDP listener for **QUIC+HTTP/3
          listen ${addr}:${toString port} http3 "
          + optionalString vhost.default "default_server "
          + optionalString vhost.http3Reuseport "reuseport"
          + ";")
```
Could be shortened to this.

---

Thanks, fixed.

---

error: syntax error, unexpected DOLLAR_CURLY, expecting ';'

       at /home/user/works/src/nixpkgs/nixos/modules/services/web-servers/nginx/default.nix:264:20:

          263|
          264|             listen ${addr}:${toString port} "
             |                    ^
          265|           + optionalString (ssl && vhost.http2) "http2 "
(use '--show-trace' to show detailed location information)